### PR TITLE
Add some eSUN Filaments

### PR DESCRIPTION
**Pull request by kazooless**
PLA+: White, Grey, Black, Purple
PETG: Black, Grey, Purple

### CODE REVIEW

**Comment by ChrisTerBeke** [comment]
Shouldn't new materials be submitted via the Marketplace?

**Comment by ChrisTerBeke** [comment]
@kazooless it's probably faster to get your material profiles released by zipping them up and uploading them via contribute.ultimaker.com. They'll be available in the Marketplace and users can install them without needing a Cura update.

**Comment by Ghostkeeper** [comment]
No, in the contributor portal we've been forbidden to accept profiles that don't support Ultimaker printers. So much for an open filament system. They said that we must add them to the Cura source instead. At first thought that doesn't make sense from Ultimaker's point of view, but built-in material profiles are hidden by Cura if the current printer doesn't support them so in the end adding them to fdm_materials results in less confusion.

Don't submit this to the Contributor portal, but rather here. It's the same people that need to review it anyway so there wouldn't be any difference in how fast we're looking at it.

**Comment by ChrisTerBeke** [comment]
Oh wow, I was not aware of that. This place it is then (with some fixing of the GUIDs).

**Comment by Ghostkeeper** [comment]
I think if we add compatibility information/filters to the Marketplace, we'll be reconsidering that.

File: eSUN_PETG_Black.xml.fdm_material [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+<fdmmaterial version="1.3" xmlns="http://www.ultimaker.com/material">
+    <metadata>
+        <name>
+            <brand>eSUN</brand>
+            <material>PETG</material>
+            <color>Black</color>
+            <label>eSUN PETG</label>
+        </name>
+        <color_code>#333333</color_code>
+        <GUID />
```
Comment (LipuFei) [inline eSUN_PETG_Black.xml.fdm_material]: `GUID` is missing.

File: eSUN_PETG_Grey.xml.fdm_material [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+<fdmmaterial version="1.3" xmlns="http://www.ultimaker.com/material">
+    <metadata>
+        <name>
+            <brand>eSUN</brand>
+            <material>PETG</material>
+            <color>Grey</color>
+            <label>eSUN PETG</label>
+        </name>
+        <color_code>#999999</color_code>
+        <GUID />
```
Comment (LipuFei) [inline eSUN_PETG_Grey.xml.fdm_material]: `GUID` is missing.

File: eSUN_PETG_Purple.xml.fdm_material [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+<fdmmaterial version="1.3" xmlns="http://www.ultimaker.com/material">
+    <metadata>
+        <name>
+            <brand>eSUN</brand>
+            <material>PETG</material>
+            <color>Purple</color>
+            <label>eSUN PETG</label>
+        </name>
+        <color_code>#9900CC</color_code>
+        <GUID />
```
Comment (LipuFei) [inline eSUN_PETG_Purple.xml.fdm_material]: `GUID` is missing.

File: eSUN_PLA_PRO_Black.xml.fdm_material [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+<fdmmaterial version="1.3" xmlns="http://www.ultimaker.com/material">
+    <metadata>
+        <name>
+            <brand>eSUN</brand>
+            <material>PLA+</material>
+            <color>Black</color>
+            <label>eSUN PLA PRO</label>
+        </name>
+        <color_code>#333333</color_code>
+        <GUID />
```
Comment (LipuFei) [inline eSUN_PLA_PRO_Black.xml.fdm_material]: `GUID` is missing.

File: eSUN_PLA_PRO_Grey.xml.fdm_material [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+<fdmmaterial version="1.3" xmlns="http://www.ultimaker.com/material">
+    <metadata>
+        <name>
+            <brand>eSUN</brand>
+            <material>PLA+</material>
+            <color>Grey</color>
+            <label>eSUN PLA PRO</label>
+        </name>
+        <color_code>#999999</color_code>
+        <GUID />
```
Comment (LipuFei) [inline eSUN_PLA_PRO_Grey.xml.fdm_material]: `GUID` is missing.

File: eSUN_PLA_PRO_Purple.xml.fdm_material [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+<fdmmaterial version="1.3" xmlns="http://www.ultimaker.com/material">
+    <metadata>
+        <name>
+            <brand>eSUN</brand>
+            <material>PLA+</material>
+            <color>Purple</color>
+            <label>eSUN PLA PRO</label>
+        </name>
+        <color_code>#9900CC</color_code>
+        <GUID />
```
Comment (LipuFei) [inline eSUN_PLA_PRO_Purple.xml.fdm_material]: `GUID` is missing.

File: eSUN_PLA_PRO_White.xml.fdm_material [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+<fdmmaterial version="1.3" xmlns="http://www.ultimaker.com/material">
+    <metadata>
+        <name>
+            <brand>eSUN</brand>
+            <material>PLA+</material>
+            <color>White</color>
+            <label>eSUN PLA PRO</label>
+        </name>
+        <color_code>#EEEEEE</color_code>
+        <GUID />
```
Comment (LipuFei) [inline eSUN_PLA_PRO_White.xml.fdm_material]: `GUID` is missing.

**Comment by kazooless** [comment]
I was not sure if the GUID was optional. How do I  generate a GUID then?

**Comment by Ghostkeeper** [comment]
import uuid
    print(uuid.uuid4())

**Comment by kazooless** [comment]
I've added the GUIDs.

**Comment by Ghostkeeper** [comment]
It all works, but if the user needs to select from the available colours he can't see beforehand which colour he chooses. It looks very weird this way.

I can't take a screenshot of these menus because Ubuntu won't let me, so here is a primitive "photograph" displaying the problem:
IMG_20191220_113551

If you leave away the `<label>` tags, then Cura will construct a label automatically based on the brand, the material and the colour. So it'll become `eSUN CPE+ Black`. And it'll be clear which material the user wants to choose.

**Comment by Ghostkeeper** [comment]
Menus are more clear now! Thanks.

**Pull request merged**